### PR TITLE
refactor: design system

### PR DESCRIPTION
**Pull request by jenshor**
The design system got a litter refactoring to make it more structured. The changes can be viewed [on Figma](https://www.figma.com/design/TQ3x8ohiB7XfUKSYeFVb7v/branch/OsLeJAAv09GZwyCWWPCR1n/FlutterCon-'24?m=auto&node-id=6809-5479&t=1B8M6TeC2N4MqCDc-1). 
To see what exactly changed click on the arrow next to _FlutterCon '24 [branch symbol] refactor_ and press _Review and merge changes..._

- added new components
- added new stories

### CODE REVIEW

File: .github/workflows/widgetbook-build.yaml [hunk 1]
```diff
@@ -1,50 +0,0 @@
...
-        uses: google-github-actions/auth@v1.0.0
-        with:
-          credentials_json: ${{ secrets.STORAGE_SERVICE_ACCOUNT }}
-
-      - name: Deploy to demo.widgetbook.io
-        uses: google-github-actions/upload-cloud-storage@v1.0.0
-        with:
-          path: widgetbook/build/web
-          destination: demo.widgetbook.io
-          parent: false
```
Comment (YoussefRaafatNasry) [inline .github/workflows/widgetbook-build.yaml:50]: **issue:** This step is still needed, to host the changes on `demo.widgetboo.io`

File: widgetbook/lib/basket/basket_summary.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import 'package:widgetbook_annotation/widgetbook_annotation.dart';
+
+@UseCase(
+  name: 'Default',
+  type: BasketSummary,
+  designLink:
+      "https://www.figma.com/design/TQ3x8ohiB7XfUKSYeFVb7v/FlutterCon-'24?node-id=6809-5479&t=Jvcq3BxVyPK6l2OF-4",
+)
+Widget buildBasketSummaryUseCase(BuildContext context) {
+  return BasketSummary(
```
Comment (YoussefRaafatNasry) [inline widgetbook/lib/basket/basket_summary.dart:13]: **question:** Should we add some padding to the use-case?

We might also need to set `mainAxisSize` to `MainAxisSize.min` inside the `BasketSummary` widget, to make the vertical centering better.

image

File: widgetbook/lib/core/app_bar.dart [hunk 3]
```diff
@@ -1,23 +1,6 @@
+// ignore_for_file: unused_import
+
 import 'package:flutter/widgets.dart';
 import 'package:groceries_app/core/core.dart';
 import 'package:widgetbook/widgetbook.dart';
 import 'package:widgetbook_annotation/widgetbook_annotation.dart';
```
Comment (YoussefRaafatNasry) [inline widgetbook/lib/core/app_bar.dart:6]: **suggestion:** I think we should keep the `AppBar` implemented in this repo, and create a `flutter-con-24-challenge` repo, with this component removed.

Comment (jenshor) [reply widgetbook/lib/core/app_bar.dart:6]: 👍🏻 I Accidentally pushed to the wrong repo

**Pull request closed without merging**